src/ngraph/runtime/allocator.cpp

```diff
@@ -51,6 +51,6 @@ class ngraph::runtime::DefaultAllocator : public ngraph::runtime::Allocator
 
 ngraph::runtime::Allocator* ngraph::runtime::get_default_allocator()
 {
-    static std::unique_ptr<DefaultAllocator> allocator(new DefaultAllocator());
-    return allocator.get();
+    static DefaultAllocator* allocator = new DefaultAllocator();
```

Comment:
It is probably a wrong solution. It used to be a singleton (though it is usually implemented without `new`s) and now it looks like a memory leak. Every time this method is called, a new allocator is created and never destroyed.

Sorry for intervention. Please let me know if it is a member only section.

Reply:
Hi @s-barannikov. Definitely not a members-only section. :)

I think this should be fine, because of the `static` qualifier.

Reply:
@aprocter 
Hi,
> I think this should be fine, because of the static qualifier.

Yes, but who is gonna call the `delete`? Anyway, the class is only 8 bytes long, so... I can't just imagine how this could solve the problem :)

Reply:
Ah, sure, I misunderstood--thought you were saying that we'd leak a new instance with every call. @pruthviIntel would this not work instead? Not sure I have complete context on what we are trying to do here.

```
static DefaultAllocator allocator;
return &allocator;
```

Reply:
@aprocter 

> thought you were saying that we'd leak a new instance with every call

It is what I wrote initially. Then I realized that static is still static and edited the comment in hope no one will read email :) Sorry for misleading